src/public/javascripts/app.js

```diff
@@ -214,17 +241,27 @@ socket.on('stats', function (data) {
 
     if (responses) {
       var deltaTime = responses.timestamp - rpsChart.data.labels[rpsChart.data.labels.length - 1];
+      if (deltaTime < 1) deltaTime = 1000;
       rpsStat.textContent = (responses.count / deltaTime * 1000).toFixed(2);
       rpsChart.data.datasets[0].data.push(responses.count / deltaTime * 1000);
       rpsChart.data.labels.push(responses.timestamp);
     }
 
+    if (responses) {
+       for(var i = 0; i < 4; i++) {
+        statusCodesChart.data.datasets[i].data.push(data.responses[i+2]);
+      }
+      statusCodesChart.data.labels.push(data.responses.timestamp);
+    }
+
     charts.forEach(function (chart) {
       if (spans[defaultSpan].retention < chart.data.labels.length) {
-        chart.data.datasets[0].data.shift();
+        chart.data.datasets.forEach((dataset) => {
+          dataset.data.shift();
+        });
+
         chart.data.labels.shift();
       }
-
       chart.update();
     });
   }
```

Comment:
in general, I think we should consider to refactor also this part of the app and decompose it into modules and use browserify to create the bundle for us; so that we can test it properly 😅